ReactiveCocoa/Swift/Signal.swift

```diff
@@ -883,6 +898,17 @@ public func combineLatest<A, B, C, D, E, F, G, H, I, J, Error>(a: Signal<A, Erro
 		|> map(repack)
 }
 
+/// Combines the values of all the given signals, in the manner described by
+/// `combineLatestWith`.
+public func combineLatest<T, Error>(signals: [Signal<T, Error>]) -> Signal<[T], Error> {
+	if let first = signals.first {
+		let initial = first |> map { [$0] }
+		return signals[1 ..< signals.count].reduce(initial) { $0 |> combineLatestWith($1) }
+	}
+	
+	return Signal.never
```

Comment:
I thought about this too and decided to solve it like this dude to the following reasons:
- a precondition would yield the function very inconvenient, since you'd always have to check whether the array is empty
- It makes sense that combining zero signals returns a signal without an event

The thing I'm a bit concerned about is whether it's obvious that when one passes an empty array to this function that the observing closures would never be called. I think I would forget about this quite easily.